fe/src/main/java/org/apache/doris/analysis/Analyzer.java

```diff
@@ -1038,7 +1038,7 @@ public boolean containsOuterJoinedTid(List<TupleId> tids) {
      */
     public SlotDescriptor getColumnSlot(TupleDescriptor tupleDesc, Column col) {
         for (SlotDescriptor slotDesc : tupleDesc.getSlots()) {
-            if (slotDesc.getColumn() == col) {
+            if (slotDesc.getColumn().equals(col)) {
```

Comment:
I think we can put this function to TupleDescriptor, and compare only with column name

Reply:
done